pkg/infra/iac2/templates_compiler.go

```diff
@@ -464,6 +464,28 @@ func (tc TemplatesCompiler) handleIaCValue(v core.IaCValue, appliedOutputs *[]Ap
 		default:
 			return "", errors.Errorf("unsupported resource type %T for '%s'", resource, property)
 		}
+	case string(core.CONNECTION_STRING):
+		switch res := v.Resource.(type) {
+		case *resources.RdsProxy:
+			downResources := tc.resourceGraph.GetUpstreamDependencies(res)
+			var instance *resources.RdsInstance
+			for _, resource := range downResources {
+				if rdsProxyTargetGroup, ok := resource.Source.(*resources.RdsProxyTargetGroup); ok {
+					instance = rdsProxyTargetGroup.RdsInstance
+				}
+			}
+			if instance == nil {
+				return "", errors.Errorf("Rds Proxy, %s, must have an associated instance", v.Resource.Id())
+			}
+
+			fetchUsername := fmt.Sprintf(`fs.readFileSync('%s', 'utf-8').split("\n")[1].split('"')[3]`, instance.CredentialsPath)
```

Comment:
It's kinda wonky, imo, that we have some of these `readFileSync`s within a factory.ts, and some righthere. And generally speaking, imo the less ts logic goes directly in go, the better.

I think this is starting to get at a "virtual resource" that I've mentioned before: a `core.Resource` that the IaC injects into the DAG, in a pass before the rendering. I know we've discussed it, but I can't remember if we're doing it yet.

In this case, it would be to identify whether we need an `RdsConnectionInfo` resource (which would be a struct within the iac package, which implements `core.Resource`). If we do, we create it — and it has a normal factory.ts that does the read, and returns a type of `{username: string, password: string}` (we could even do it as an async read, and return a `Promise<{username: string, password: string}>`).

Reply:
yeah i mean thats the other route we could go, is to create some resource to move this out, but we dont do that yet. Would you be ok with me cutting a ticket, specifically calling this one out and moving that to fit and finish? Since this works im wondering if it could be a checkpoint, but for now because this has to be in sync with the env var, and the secret i dont want to make this pr too much larger